### PR TITLE
Challenge 2

### PR DESCRIPTION
**Pull request by machariamarigi**
**What does this PR do?**
The completion of Andela Bootcamp 21 Challenge 2

**Description of the task Completed**

1. The application was integrated with Flask.
2. Test Driven Development and Continous Integration were introduced
3. Application was deployed to Heroku => https://shopping-list-flasky.herokuapp.com/
4. Various features were implemented using OOP and Python data structures.

- Users can register to use the app
- Registered users can log into the app
- Logged in users can Create, Update and Delete shopping lists
- Logged in users can Create, Update and Delete items into shopping lists
- Logged in users can log out

**How to Run the pplicatiom**
After cloning the Repo cd into it and install all requirements, run commands `export FLASK_CONFIG="development"` and `python run.py` to run the application

### CODE REVIEW

**Comment by xcixor** [comment]
Looking good bro, kudos

**Comment by machariamarigi** [comment]
Thanks @xcixor

**Comment by Chris-Maina** [comment]
Great work.

File: tests/test_models.py [hunk 1]
```diff
@@ -94,15 +94,14 @@ def test_add_shoppinglist(self):
     def test_get_shoppinglist(self):
         """Test whether we can get a single shoppinglist"""
         self.test_store.add_user('mash', 'mash@mash2.com', 'mash_pass')
-        self.test_store.add_shoppinglist(3, "hardware")
-        test_item = self.test_store.get_shoppinglist(3, 1)
+        self.test_store.add_shoppinglist(2, "hardware")
+        test_item = self.test_store.get_shoppinglist(2, 1)
         self.assertEquals(
```
Comment (HawiCaesar) [inline tests/test_models.py]: This is okay and still works. As you move forward you may want to consider using self.assertEqual for comparing numbers mostly and not so much of other data structures. You can using self.assertIn to check if a dictionary key exists in the dictionary.

**Comment by HawiCaesar** [comment]
This is good work.
Just a question. I managed to register into you app. When I did I was taken directly to my shopping list page. Why not to login page to actually now login?

File: tests/test_models.py (hunk 1 above)
Comment (machariamarigi) [reply tests/test_models.py]: Thank you for the suggestion, makes total sense. I'll start looking into applying this.

**Comment by machariamarigi** [comment]
@HawiCaeser I made that UX decession since I think a user would appreciate registering and starting to use app directly rather than having the extra step of logging in. Users can still log out and log in again

**Pull request merged**